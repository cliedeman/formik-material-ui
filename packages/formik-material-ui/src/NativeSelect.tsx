import * as React from 'react';
import MuiNativeSelect, {
  NativeSelectProps as MuiNativeSelectProps,
} from '@material-ui/core/NativeSelect';
import { FieldProps } from 'formik';

export interface NativeSelectProps
  extends FieldProps,
    Omit<MuiNativeSelectProps, 'name' | 'value'> {}

export function fieldToSelect({
  disabled,
  field: { onBlur: fieldOnBlur, ...field },
  form: { isSubmitting },
  onBlur,
  ...props
}: NativeSelectProps): MuiNativeSelectProps {
  return {
    disabled: disabled ?? isSubmitting,
    onBlur:
      onBlur ??
      function (e) {
        fieldOnBlur(e ?? field.name);
      },
    ...field,
    ...props,
  };
}

export function Select(props: NativeSelectProps) {
  return <MuiNativeSelect {...fieldToSelect(props)} />;
}

Select.displayName = 'FormikMaterialUISelect';
