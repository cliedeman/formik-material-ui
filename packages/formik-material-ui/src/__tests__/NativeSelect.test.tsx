import * as React from 'react';
import { test, expect } from '@jest/globals';
import { Formik, Form, Field } from 'formik';
import renderer from 'react-test-renderer';

import { NativeSelect } from '../NativeSelect';

test('NativeSelect Renders Correctly', () => {
  const component = renderer.create(
    <Formik onSubmit={() => {}} initialValues={{ test: '' }}>
      <Form>
        <Field component={NativeSelect} name="test">
          <option value="">
            <em>None</em>
          </option>
          <option value={10}>Ten</option>
          <option value={20}>Twenty</option>
          <option value={30}>Thirty</option>
        </Field>
      </Form>
    </Formik>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
