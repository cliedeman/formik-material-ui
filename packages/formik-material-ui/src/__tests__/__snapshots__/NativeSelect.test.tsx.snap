// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Renders Correctly 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline"
    onClick={[Function]}
  >
    <select
      className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
      disabled={false}
      name="test"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value=""
    >
      <option
        value=""
      >
        <em>
          None
        </em>
      </option>
      <option
        value={10}
      >
        Ten
      </option>
      <option
        value={20}
      >
        Twenty
      </option>
      <option
        value={30}
      >
        Thirty
      </option>
    </select>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiNativeSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</form>
`;
